### PR TITLE
vcpkg integrate: Add support for zsh

### PR DESCRIPTION
**Pull request by autoantwort**
Closes microsoft/vcpkg#17742

### CODE REVIEW

File: src/vcpkg/commands.integrate.cpp [hunk 1]
```diff
@@ -482,6 +482,67 @@ With a project open, go to Tools->NuGet Package Manager->Package Manager Console
...
+        std::vector<std::string> zshrc_content = maybe_zshrc_content.value_or_exit(VCPKG_LINE_INFO);
+
+        // How to use bash completions in zsh: https://stackoverflow.com/a/8492043/10162645
+        bool has_autoload_bashcompinit = false;
+        bool has_bashcompinit = false;
+        std::vector<std::string> matches;
+        for (auto&& line : zshrc_content)
+        {
+            std::smatch match;
+            if (std::regex_match(line, match, std::regex{R"###(^source.*scripts/vcpkg_completion.zsh$)###"}))
```
Comment (BillyONeal) [inline src/vcpkg/commands.integrate.cpp:506]: The ^$ dance is already implied by regex_match as opposed to regex_search I think. Given the missing $ on 510 it seems like that might not do what you want?

Comment (autoantwort) [reply src/vcpkg/commands.integrate.cpp:506]: Yeah right. I have copied from `integrate_bash`, so I have also fixed it there

Comment (BillyONeal) [reply src/vcpkg/commands.integrate.cpp]: > I have copied from `integrate_bash`, so I have also fixed it there

Ouch

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
This looks good from a vcpkg view; I'm trying to get someone who understands zsh to help review https://twitter.com/MalwareMinigun/status/1415763885441765377

**Comment by ras0219-msft** [comment]
Since there's a lack of progress on getting additional zsh reviewers, should this be merged?

**Comment by autoantwort** [comment]
If you ask me, yes ;) I am using it on my mac with zsh and it works :)

**Comment by BillyONeal** [comment]
`¯\_(ツ)_/¯` Thanks for your contribution!

**Pull request merged**